ui/raidboss/raidboss_config.ts

```diff
@@ -1440,17 +1438,15 @@ const flattenTimeline = (
 const userFileHandler: UserFileCallback = (
   name: string,
   files: { [filename: string]: string },
-  baseOptions: BaseOptions,
+  baseOptions: (BaseOptions & Partial<RaidbossOptions>),
```

Comment:
This is more of a stop-gap until the TODO comment below can be handled, but it's more accurate than the `as` cast which was here before.

Reply:
I think this isn't true in practice, though.  I think the options really is a full `RaidbossOptions` and not a partial one.  There's default values for all fields.

I don't think this was true in the past, pre-typescript.  But it is true now, e.g. the default export from `raidboss_options.ts`.

Reply:
Typescript doesn't like it which is why I didn't go for it to begin with.
![image](https://user-images.githubusercontent.com/6119598/168204543-5c86b4d2-70f9-4ca6-8819-be3212bdcea6.png)
(same error if type is `BaseOptions & RaidbossOptions`)

Reply:
Yeah, agreed that you can't do this without a type assertion.  I was just saying that it *is* a `RaidbossOptions`, regardless of what TypeScript knows about it.

Ok, I think removing the type assertion is fine here and we can clean up the conditionals if this TODO ever gets fixed.